services/core-api/app/api/now_applications/resources/administrative_amendment_list_resource.py

```diff
@@ -0,0 +1,180 @@
+                for mine_appt in [
+                        party for party in permit.permittee_appointments if not party.end_date
+                        or party.end_date >= datetime.now(timezone.utc).date()
+                ]:
+                    new_app_appt = NOWPartyAppointment(
+                        mine_party_appt_type_code=mine_appt.mine_party_appt_type_code,
+                        now_application_id=new_app.now_application_id,
+                        party_guid=mine_appt.party_guid)
+
+                    db.session.add(new_app_appt)
+                    application_appt.append(new_app_appt)
+                new_app.contacts = application_appt
+
+            def get_documents_to_attach(db, documents):
+                res_documents = []
+                for doc in [doc for doc in documents if doc.is_final_package]:
+                    mine_doc = MineDocument(
+                        document_name=doc.mine_document.document_name,
+                        document_manager_guid=doc.mine_document.document_manager_guid,
+                        mine_guid=mine.mine_guid)
+                    db.session.add(mine_doc)
+
+                    doc_type_code = doc.now_application_document_type_code if hasattr(
+                        doc, 'now_application_document_type_code') else "OTH"
+
+                    new_appt_doc = NOWApplicationDocumentXref(
+                        mine_document=mine_doc,
+                        now_application_id=new_app.now_application_id,
+                        is_final_package=doc.is_final_package,
+                        preamble_title=doc.preamble_title,
+                        preamble_author=doc.preamble_author,
+                        preamble_date=doc.preamble_date,
+                        mine_document_guid=doc.mine_document_guid,
+                        description=doc.description,
+                        now_application_document_type_code=doc_type_code)
+
+                    db.session.add(new_appt_doc)
+
+                    res_documents.append(doc)
+                return res_documents
+
+            if application:
+                documents_to_copy = []
+
+                if application.documents:
+                    documents_to_copy.append(get_documents_to_attach(db, application.documents))
+
+                if application.imported_submission_documents:
+                    documents_to_copy.append(
+                        get_documents_to_attach(db, application.imported_submission_documents))
+
+                new_app.documents = documents_to_copy
+
+            for reason in data['application_reason_codes']:
+                app_reason = ApplicationReasonXref(
+                    now_application_id=new_app.now_application_id, application_reason_code=reason)
+                db.session.add(app_reason)
+
+            db.session.add(new_app)
+            db.session.commit()
+        except:
+            db.session.rollback()
+            raise
```

Comment:
We should specify what we are raising here?

Reply:
At this point I'm reraising the original exception, I'm using this catch as rollback point and allowing all our other mechanisms to handle the exception